browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,474 @@
+#include "content/public/browser/global_routing_id.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/renderer/render_frame.h"
+#include "brave/components/cosmetic_filters/resources/grit/cosmetic_filters_generated_map.h"
+#include "ui/base/resource/resource_bundle.h"
+
+const char k_pre_init_script[] =
+  "(function() {"
+     "if (window.content_cosmetic == undefined) {"
+        "window.content_cosmetic = {};"
+     "}"
+     "%s"
+     "%s"
+  "})();";
+
+const char k_scriplet_init_script[] =
+  "if (window.content_cosmetic.scriplet == undefined ||"
+      "window.content_cosmetic.scriplet === '') {"
+    "let text = %s;"
+    "window.content_cosmetic.scriplet = `${text}`;"
+  "}";
+
+const char k_non_scriplet_init_script[] =
+  "if (window.content_cosmetic.hide1pContent === undefined) {"
+      "window.content_cosmetic.hide1pContent = %s;"
+  "}"
+  "if (window.content_cosmetic.generichide === undefined) {"
+      "window.content_cosmetic.generichide = %s;"
+  "}";
+
+const char k_selectors_inject_script[] =
+  "(function() {"
+     "let nextIndex ="
+        "window.content_cosmetic.cosmeticStyleSheet.rules.length;"
+     "const selectors = %s;"
+     "selectors.forEach(selector => {"
+       "if (!window.content_cosmetic.allSelectorsToRules.has(selector)) {"
+         "let rule = selector + '{display:none !important;}';"
+         "window.content_cosmetic.cosmeticStyleSheet.insertRule("
+           "`${rule}`, nextIndex);"
+         "window.content_cosmetic.allSelectorsToRules.set("
+           "selector, nextIndex);"
+         "nextIndex++;"
+         "window.content_cosmetic.firstRunQueue.add(selector);"
+       "}"
+     "});"
+     "if (!document.adoptedStyleSheets.includes("
+         "window.content_cosmetic.cosmeticStyleSheet)) {"
+       "document.adoptedStyleSheets ="
+         "[window.content_cosmetic.cosmeticStyleSheet];"
+     "};"
+  "})();";
+
+const char k_style_selectors_inject_script[] =
+  "(function() {"
+    "let nextIndex ="
+        "window.content_cosmetic.cosmeticStyleSheet.rules.length;"
+    "const selectors = %s;"
+    "for (let selector in selectors) {"
+      "if (!window.content_cosmetic.allSelectorsToRules.has(selector)) {"
+        "let rule = selector + '{';"
+        "selectors[selector].forEach(prop => {"
+          "if (!rule.endsWith('{')) {"
+            "rule += ';';"
+          "}"
+          "rule += prop;"
+        "});"
+        "rule += '}';"
+        "window.content_cosmetic.cosmeticStyleSheet.insertRule("
+          "`${rule}`, nextIndex);"
+        "window.content_cosmetic.allSelectorsToRules.set("
+          "selector, nextIndex);"
+        "nextIndex++;"
+      "};"
+    "};"
+    "if (!document.adoptedStyleSheets.includes("
+          "window.content_cosmetic.cosmeticStyleSheet)){"
+       "document.adoptedStyleSheets ="
+         "[window.content_cosmetic.cosmeticStyleSheet];"
+    "};"
+  "})();";
+
+std::vector<std::string>
```

Comment:
I think these have fallen out of sync at least Yahoo, and maybe others, should be removed i believe